homeassistant/components/evohome.py

```diff
@@ -0,0 +1,690 @@
+                "tcs._set_status(%s)...",
+                operation_mode
+            )
+
+            try:
+                self._obj._set_status(operation_mode)                           # noqa: E501; pylint: disable=protected-access
+            except requests.exceptions.HTTPError as err:
+                self._handle_requests_exceptions("HTTPError", err)
+
+        else:
+            raise NotImplementedError()
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        _LOGGER.debug("turn_away_mode_on(%s)", self._id)
+        self.set_operation_mode(EVO_AWAY)
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        _LOGGER.debug("turn_away_mode_off(%s)", self._id)
+        self.set_operation_mode(EVO_AUTO)
+
+    def _update_state_data(self, domain_data):
+        client = domain_data['client']
+        loc_idx = domain_data['params'][CONF_LOCATION_IDX]
+
+        # Obtain latest state data (e.g. temps)...
+        _LOGGER.debug(
+            "_update_state_data(): API call [1 request(s)]: "
+            "client.locations[loc_idx].status()..."
+        )
+
+        try:
+            domain_data['status'].update(  # or: domain_data['status'] =
+                client.locations[loc_idx].status()[GWS][0][TCS][0])
+
+        except requests.exceptions.HTTPError as err:
+            self._handle_requests_exceptions("HTTPError", err)
+
+        else:
+            # only update the timers if the api call was successful
+            domain_data['timers']['statusUpdated'] = datetime.now()
+
+        _LOGGER.debug(
+            "_update_state_data(): domain_data['status'] = %s",
+            domain_data['status']
+        )
+
+    def update(self):
+        """Get the latest state data of the installation.
+
+        This includes state data for the Controller and its slave devices, such
+        as the operating_mode of the Controller and the current_temperature
+        of slaves.
+
+        This is not asyncio-friendly due to the underlying client api.
+        """
+        domain_data = self.hass.data[DATA_EVOHOME]
+
+        # Wait a minimum (scan_interval/60) mins (rounded up) between updates
+        timeout = datetime.now() + timedelta(seconds=55)
+        expired = timeout > self._timers['statusUpdated'] + \
+            timedelta(seconds=domain_data['params'][CONF_SCAN_INTERVAL])
+
+        if not expired:  # timer not expired, so exit
+            return True
+
+# it is time to update state data
+# NB: unlike all other config/state data, zones maintain their own schedules
+        self._update_state_data(domain_data)
+        self._status = domain_data['status']
+
+        _LOGGER.debug(
+            "update(%s), self._status = %s",
+            self._id,
+            self._status
+        )
+
+        return True
+
+    @property
+    def target_temperature(self):
+        """Return the average target temperature of the Heating/DHW zones."""
+        num_temps = total_temp = 0
```

Comment:
```python
temps = [zone['setpointStatus']['targetHeatTemperature'] 
         for zone in self._status['zones']]

avg_temp = sum(temps) / len(temps) if len(temps) > 0 else None
```

Reply:
I never used Python before this project - every day I lean something new!

Done